src/extension/networkInspector/requestBodyFormatters/requestBodyFormatter.ts

```diff
@@ -0,0 +1,109 @@
+import { notNullOrUndefined } from "../../../common/utils";
+
+export interface IFormatter {
+    formatRequest?: (request: Request, contentType: string) => string | any | null;
+    formatResponse?: (response: Response, contentType: string) => string | any | null;
+}
+
+export class RequestBodyFormatter {
+    protected logger: OutputChannelLogger;
+    private formatters: Array<IFormatter>;
+
+    constructor(logger: OutputChannelLogger) {
+        this.logger = logger;
+        this.formatters = [
+            new ImageFormatter(),
+            new GraphQLFormatter(this.logger),
+            new JSONFormatter(this.logger),
+            new FormUrlencodedFormatter(this.logger),
+        ];
+    }
+
+    public formatBody(container: Request | Response): string | any {
+        const contentType = getHeaderValue(container.headers, "content-type");
+
+        for (let formatter of this.formatters) {
+            try {
+                let formattedRes = null;
+                // if container is a response
+                if ((<any>container).status) {
+                    if (formatter.formatResponse) {
+                        formattedRes = formatter.formatResponse(<Response>container, contentType);
+                    }
+                } else if (formatter.formatRequest) {
+                    formattedRes = formatter.formatRequest(<Request>container, contentType);
+                }
+
+                if (notNullOrUndefined(formattedRes)) {
+                    return formattedRes;
+                }
+            } catch (err) {
+                this.logger.debug(
+                    `RequestBodyFormatter exception from ${formatter.constructor.name} ${err.message}`,
+                );
+            }
+        }
+
+        return decodeBody(container, this.logger);
+    }
+}
+
+export function decodeBody(container: Request | Response, logger?: OutputChannelLogger): string {
+    if (!container.data) {
+        return "";
+    }
+
+    try {
+        const isGzip = getHeaderValue(container.headers, "Content-Encoding") === "gzip";
+        if (isGzip) {
+            try {
+                const binStr = Base64.atob(container.data);
+                const dataArr = new Uint8Array(binStr.length);
+                for (let i = 0; i < binStr.length; i++) {
+                    dataArr[i] = binStr.charCodeAt(i);
+                }
+                // The request is gzipped, so convert the base64 back to the raw bytes first,
+                // then inflate. pako will detect the BOM headers and return a proper utf-8 string right away
+                return pako.inflate(dataArr, { to: "string" });
+            } catch (e) {
+                // on iOS, the stream send to flipper is already inflated, so the content-encoding will not
+                // match the actual data anymore, and we should skip inflating.
+                // In that case, we intentionally fall-through
+                if (!("" + e).includes("incorrect header check")) {
```

Comment:
Maybe it's better to do type casting by using `as string`, `<string>`, or `.toString()` here?